src/persist-client/src/internal/compact.rs

```diff
@@ -177,6 +184,7 @@ impl Compactor {
         metrics: Arc<Metrics>,
         cpu_heavy_runtime: Arc<CpuHeavyRuntime>,
         req: CompactReq<T>,
+        memory_budget_bytes: usize,
```

Comment:
I have a reasonably strong preference that this should instead be a field on PersistConfig. tests that need to override the default can clone PersistConfig and mutate it as necessary